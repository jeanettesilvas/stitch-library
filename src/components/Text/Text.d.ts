/// <reference types="react" />
import './Text.scss';
interface TextProps {
    children: any;
    className?: string;
    element?: 'p' | 'div';
    html?: string;
    type?: 'callout';
}
export declare const Text: (props: TextProps) => JSX.Element;
export {};
