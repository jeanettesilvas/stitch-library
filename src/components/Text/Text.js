import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import './Text.scss'

export class Text extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    element: PropTypes.oneOf(['p', 'div']),
    type: PropTypes.oneOf(['callout']),
  }

  static defaultProps = {
    element: 'p',
  }

  render() {
    const classNames = classnames('text', this.props.className, {
      'text--callout': this.props.type === 'callout',
    })

    return (
      <this.props.element className={classNames}>
        {this.props.children}
      </this.props.element>
    )
  }
}
