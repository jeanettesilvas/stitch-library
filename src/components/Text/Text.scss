@import '../../style/global.scss';

.text {
  color: $gray-5;
  font-size: 16px;
  font-family: $base-font;
  line-height: 26px;
  text-align: left;

  &--callout {
    font-size: 18px;
    line-height: 30px;
  }
}
