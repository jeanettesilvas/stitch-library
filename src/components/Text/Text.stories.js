import React from 'react'
import { storiesOf } from '@storybook/react'
import { Text } from './Text'

storiesOf('Text', module)
  .addDecorator(story => <div style={{ padding: '50px' }}>{story()}</div>)
  .add('default', () => (
    <Text>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Diam phasellus
      vestibulum lorem sed risus. Elementum sagittis vitae et leo duis ut.
      Bibendum enim facilisis gravida neque convallis a. Morbi non arcu risus
      quis varius. Egestas tellus rutrum tellus pellentesque eu tincidunt tortor
      aliquam. Iaculis nunc sed augue lacus viverra vitae congue eu. Id ornare
      arcu odio ut sem nulla pharetra. Facilisis volutpat est velit egestas dui
      id. Eu nisl nunc mi ipsum. Proin sagittis nisl rhoncus mattis rhoncus urna
      neque viverra. Cras tincidunt lobortis feugiat vivamus at augue eget arcu
      dictum. Massa tempor nec feugiat nisl pretium fusce id velit. Ornare
      aenean euismod elementum nisi quis eleifend. Metus dictum at tempor
      commodo ullamcorper.
    </Text>
  ))
  .add('callout', () => (
    <Text element="div" type="callout">
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
      tempor incididunt ut labore et dolore magna aliqua. Diam phasellus
      vestibulum lorem sed risus. Elementum sagittis vitae et leo duis ut.
      Bibendum enim facilisis gravida neque convallis a. Morbi non arcu risus
      quis varius.
    </Text>
  ))
