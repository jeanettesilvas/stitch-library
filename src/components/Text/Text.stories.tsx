import React from 'react'
import { Story, Meta } from '@storybook/react'

import { Text, TextProps } from './Text'

export default {
  title: 'Components/Text',
  component: Text,
} as Meta

const Template: Story<TextProps> = (args) => (
  <Text {...args}>{args.children}</Text>
)

export const Basic = Template.bind({})
Basic.args = {
  children: 'Hello',
  element: 'p',
}
