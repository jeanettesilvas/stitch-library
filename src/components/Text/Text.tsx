import classnames from 'classnames'

import './Text.scss'

export interface TextProps {
  children?: JSX.Element | string
  className?: string
  element?: 'p' | 'div'
  html?: string
  type?: 'callout'
}

export const Text = (props: TextProps) => {
  let { children, className, element = 'p', html = '', type } = props

  const classNames = classnames('text', className, {
    'text--callout': type === 'callout',
  })

  return (
    <>
      {(html && element === 'div') ?? (
        <div
          className={classNames}
          dangerouslySetInnerHTML={{ __html: html }}
        />
      )}
      {html && element === 'p' && <p className={classNames}>{children}</p>}
      {!html && <span className={classNames}>{children}</span>}
    </>
  )
}
