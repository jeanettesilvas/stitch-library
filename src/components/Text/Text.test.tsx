import renderer from 'react-test-renderer'

import { Text } from './Text'

describe('Text', () => {
  it('renders', () => {
    const tree = renderer.create(<Text>hello</Text>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
