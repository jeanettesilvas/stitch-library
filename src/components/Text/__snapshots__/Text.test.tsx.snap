// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders 1`] = `
Array [
  "",
  "",
  <span
    className="text"
  >
    hello
  </span>,
]
`;
